File: test-services/ts/template/src/services/echo.ts
import * as service from '../spec/echo'
import * as models from '../spec/models'

export let echoService = () => {
    let echoBodyString = async (params: service.EchoBodyStringParams): Promise<string> => {
        return params.body
    }

    let echoBodyModel = async (params: service.EchoBodyModelParams): Promise<models.Message> => {
        return {int_field: params.body.int_field, string_field: params.body.string_field}
    }

    let echoBodyArray = async (params: service.EchoBodyArrayParams): Promise<string[]> => {
        return params.body
    }

    let echoBodyMap = async (params: service.EchoBodyMapParams): Promise<Record<string, string>> => {
        return params.body
    }

    let echoFormData = async (params: service.EchoFormDataParams): Promise<models.FormParameters> => {
        return {
            int_field: params.intParam,
            long_field: params.longParam,
            float_field: params.floatParam,
            double_field: params.doubleParam,
            decimal_field: params.decimalParam,
            bool_field: params.boolParam,
            string_field: params.stringParam,
            string_opt_field: params.stringOptParam,
            string_defaulted_field: params.stringDefaultedParam,
            string_array_field: params.stringArrayParam,
            uuid_field: params.uuidParam,
            date_field: params.dateParam,
            date_array_field: params.dateArrayParam,
            datetime_field: params.datetimeParam,
            enum_field: params.enumParam,
        }
    }

    let echoFormUrlencoded = async (params: service.EchoFormUrlencodedParams): Promise<models.FormParameters> => {
        return {
            int_field: params.intParam,
            long_field: params.longParam,
            float_field: params.floatParam,
            double_field: params.doubleParam,
            decimal_field: params.decimalParam,
            bool_field: params.boolParam,
            string_field: params.stringParam,
            string_opt_field: params.stringOptParam,
            string_defaulted_field: params.stringDefaultedParam,
            string_array_field: params.stringArrayParam,
            uuid_field: params.uuidParam,
            date_field: params.dateParam,
            date_array_field: params.dateArrayParam,
            datetime_field: params.datetimeParam,
            enum_field: params.enumParam,
        }
    }

    let echoQuery = async (params: service.EchoQueryParams): Promise<models.Parameters> => {
        return {
            int_field: params.intQuery,
            long_field: params.longQuery,
            float_field: params.floatQuery,
            double_field: params.doubleQuery,
            decimal_field: params.decimalQuery,
            bool_field: params.boolQuery,
            string_field: params.stringQuery,
            string_opt_field: params.stringOptQuery,
            string_defaulted_field: params.stringDefaultedQuery,
            string_array_field: params.stringArrayQuery,
            uuid_field: params.uuidQuery,
            date_field: params.dateQuery,
            date_array_field: params.dateArrayQuery,
            datetime_field: params.datetimeQuery,
            enum_field: params.enumQuery,
        }
    }

    let echoHeader = async (params: service.EchoHeaderParams): Promise<models.Parameters> => {
        return {
            int_field: params.intHeader,
            long_field: params.longHeader,
            float_field: params.floatHeader,
            double_field: params.doubleHeader,
            decimal_field: params.decimalHeader,
            bool_field: params.boolHeader,
            string_field: params.stringHeader,
            string_opt_field: params.stringOptHeader,
            string_defaulted_field: params.stringDefaultedHeader,
            string_array_field: params.stringArrayHeader,
            uuid_field: params.uuidHeader,
            date_field: params.dateHeader,
            date_array_field: params.dateArrayHeader,
            datetime_field: params.datetimeHeader,
            enum_field: params.enumHeader,
        }
    }

    let echoUrlParams = async (params: service.EchoUrlParamsParams): Promise<models.UrlParameters> => {
        return {
            int_field: params.intUrl,
            long_field: params.longUrl,
            float_field: params.floatUrl,
            double_field: params.doubleUrl,
            decimal_field: params.decimalUrl,
            bool_field: params.boolUrl,
            string_field: params.stringUrl,
            uuid_field: params.uuidUrl,
            date_field: params.dateUrl,
            datetime_field: params.datetimeUrl,
            enum_field: params.enumUrl,
        }
    }

    let echoEverything = async (params: service.EchoEverythingParams): Promise<service.EchoEverythingResponse> => {
        return {
            status: "ok",
            data: {
                body_field: params.body,
                float_query: params.floatQuery,
                bool_query: params.boolQuery,
                uuid_header: params.uuidHeader,
                datetime_header: params.datetimeHeader,
                date_url: params.dateUrl,
                decimal_url: params.decimalUrl,
            }
        }
    }

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }

    return {echoBodyString, echoBodyModel, echoBodyArray, echoBodyMap, echoFormData, echoFormUrlencoded, echoQuery, echoHeader, echoUrlParams, echoEverything, sameOperationName}
}

File: test-services/go/template/services/echo.go
package services

import (
	"the-service/spec/echo"
	"the-service/spec/empty"
	"the-service/spec/models"

	"cloud.google.com/go/civil"
	"github.com/google/uuid"
	"github.com/shopspring/decimal"
)

type EchoService struct{}

func (service *EchoService) EchoBodyString(body string) (*string, error) {
	return &body, nil
}
func (service *EchoService) EchoBodyModel(body *models.Message) (*models.Message, error) {
	return body, nil
}
func (service *EchoService) EchoBodyArray(body *[]string) (*[]string, error) {
	return body, nil
}
func (service *EchoService) EchoBodyMap(body *map[string]string) (*map[string]string, error) {
	return body, nil
}
func (service *EchoService) EchoFormData(intParam int, longParam int64, floatParam float32, doubleParam float64, decimalParam decimal.Decimal, boolParam bool, stringParam string, stringOptParam *string, stringDefaultedParam string, stringArrayParam []string, uuidParam uuid.UUID, dateParam civil.Date, dateArrayParam []civil.Date, datetimeParam civil.DateTime, enumParam models.Choice) (*models.FormParameters, error) {
	return &models.FormParameters{IntField: intParam, LongField: longParam, FloatField: floatParam, DoubleField: doubleParam, DecimalField: decimalParam, BoolField: boolParam, StringField: stringParam, StringOptField: stringOptParam, StringDefaultedField: stringDefaultedParam, StringArrayField: stringArrayParam, UuidField: uuidParam, DateField: dateParam, DateArrayField: dateArrayParam, DatetimeField: datetimeParam, EnumField: enumParam}, nil
}
func (service *EchoService) EchoFormUrlencoded(intParam int, longParam int64, floatParam float32, doubleParam float64, decimalParam decimal.Decimal, boolParam bool, stringParam string, stringOptParam *string, stringDefaultedParam string, stringArrayParam []string, uuidParam uuid.UUID, dateParam civil.Date, dateArrayParam []civil.Date, datetimeParam civil.DateTime, enumParam models.Choice) (*models.FormParameters, error) {
	return &models.FormParameters{IntField: intParam, LongField: longParam, FloatField: floatParam, DoubleField: doubleParam, DecimalField: decimalParam, BoolField: boolParam, StringField: stringParam, StringOptField: stringOptParam, StringDefaultedField: stringDefaultedParam, StringArrayField: stringArrayParam, UuidField: uuidParam, DateField: dateParam, DateArrayField: dateArrayParam, DatetimeField: datetimeParam, EnumField: enumParam}, nil
}
func (service *EchoService) EchoQuery(intQuery int, longQuery int64, floatQuery float32, doubleQuery float64, decimalQuery decimal.Decimal, boolQuery bool, stringQuery string, stringOptQuery *string, stringDefaultedQuery string, stringArrayQuery []string, uuidQuery uuid.UUID, dateQuery civil.Date, dateArrayQuery []civil.Date, datetimeQuery civil.DateTime, enumQuery models.Choice) (*models.Parameters, error) {
	return &models.Parameters{IntField: intQuery, LongField: longQuery, FloatField: floatQuery, DoubleField: doubleQuery, DecimalField: decimalQuery, BoolField: boolQuery, StringField: stringQuery, StringOptField: stringOptQuery, StringDefaultedField: stringDefaultedQuery, StringArrayField: stringArrayQuery, UuidField: uuidQuery, DateField: dateQuery, DateArrayField: dateArrayQuery, DatetimeField: datetimeQuery, EnumField: enumQuery}, nil
}
func (service *EchoService) EchoHeader(intHeader int, longHeader int64, floatHeader float32, doubleHeader float64, decimalHeader decimal.Decimal, boolHeader bool, stringHeader string, stringOptHeader *string, stringDefaultedHeader string, stringArrayHeader []string, uuidHeader uuid.UUID, dateHeader civil.Date, dateArrayHeader []civil.Date, datetimeHeader civil.DateTime, enumHeader models.Choice) (*models.Parameters, error) {
	return &models.Parameters{IntField: intHeader, LongField: longHeader, FloatField: floatHeader, DoubleField: doubleHeader, DecimalField: decimalHeader, BoolField: boolHeader, StringField: stringHeader, StringOptField: stringOptHeader, StringDefaultedField: stringDefaultedHeader, StringArrayField: stringArrayHeader, UuidField: uuidHeader, DateField: dateHeader, DateArrayField: dateArrayHeader, DatetimeField: datetimeHeader, EnumField: enumHeader}, nil
}
func (service *EchoService) EchoUrlParams(intUrl int, longUrl int64, floatUrl float32, doubleUrl float64, decimalUrl decimal.Decimal, boolUrl bool, stringUrl string, uuidUrl uuid.UUID, dateUrl civil.Date, datetimeUrl civil.DateTime, enumUrl models.Choice) (*models.UrlParameters, error) {
	return &models.UrlParameters{IntField: intUrl, LongField: longUrl, FloatField: floatUrl, DoubleField: doubleUrl, DecimalField: decimalUrl, BoolField: boolUrl, StringField: stringUrl, UuidField: uuidUrl, DateField: dateUrl, DatetimeField: datetimeUrl, EnumField: enumUrl}, nil
}
func (service *EchoService) EchoEverything(body *models.Message, floatQuery float32, boolQuery bool, uuidHeader uuid.UUID, datetimeHeader civil.DateTime, dateUrl civil.Date, decimalUrl decimal.Decimal) (*echo.EchoEverythingResponse, error) {
	return &echo.EchoEverythingResponse{Ok: &models.Everything{BodyField: *body, FloatQuery: floatQuery, BoolQuery: boolQuery, UuidHeader: uuidHeader, DatetimeHeader: datetimeHeader, DateUrl: dateUrl, DecimalUrl: decimalUrl}}, nil
}
func (service *EchoService) SameOperationName() (*echo.SameOperationNameResponse, error) {
	return &echo.SameOperationNameResponse{Ok: &empty.Value}, nil
}
